text/0000-lifecycle-multicall-binary-for-build.md

```diff
@@ -0,0 +1,78 @@
+# Meta
+[meta]: #meta
+- Name: Lifecycle as a multicall binary for build phases
+- Start Date: 01/21/2020
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+
+We should compile the lifecycle as a multicall binary for the build phases of detect, analyze, restore, build, export and rebase while retaining the interface to the phases. The binary can also support the phases as subcommands with the phases in their verb form.
+
+
+# Motivation
+[motivation]: #motivation
+
+- The size of the lifecycle release decreases from 24M to 7M.
```

Comment:
I know we've gotten complaints about the size of images, is there data or something from slack we can point to as evidence here? It'd be great for historical purposes when people come back to this RFC.

Reply:
This RFC was originally a request from @dgageot in buildpacks/lifecycle#191